### PR TITLE
[Docs] Elaborated on alternatives to elimination contexts

### PR DESCRIPTION
**Pull request by effectfully**

### CODE REVIEW

File: docs/fomega/deep-isorecursive/alternatives.md [hunk 1]
```diff
@@ -16,19 +16,116 @@ Also iterated application is common at the type level as it simplifies unificati
...
+succ = \(n : nat) -> wrap natF [] /\(r :: *) -> \(z : r) (f : nat -> r) -> f n
+```
+
+the `list` data type
+
+```
+listF = \(list :: * -> *) (a :: *) -> all (r :: *). r -> (a -> list a -> r) -> r
+list  = \(a :: *) -> fix listF [a]
+nil   = /\(a :: *) -> wrap listF [a] /\(r :: *) -> \(z : r) (f : a -> list a -> r) -> z
+cons  = /\(a :: *) -> \(x : a) (xs : list a) ->
```
Comment (vmchale) [inline docs/fomega/deep-isorecursive/alternatives.md]: What's the syntax here? I've never seen anyone use the `/\`. Do you have any references to the literature?

Comment (effectfully) [reply docs/fomega/deep-isorecursive/alternatives.md]: This is a fairly standard notation. See for example the "16.1 Polymorphic Abstraction" chapter of [Practical Foundations for Programming Languages](https://www.cs.cmu.edu/~rwh/pfpl/2nded.pdf). `/\` is the ASCII version of `Λ`, a big lambda. People sometimes call type abstractions big lambdas (and I do this all the time).

Comment (michaelpj) [reply docs/fomega/deep-isorecursive/alternatives.md]: Yeah, this is standard AFAIK. GHC Core uses it too, it's been used by every version of System F I've seen.

Comment (effectfully) [reply docs/fomega/deep-isorecursive/alternatives.md]: I've just checked, TAPL uses `λ` for both the kinds of lambdas.

**Comment by michaelpj** [comment]
LGTM. I hadn't realised the need for passing dummy type arguments for non-parameterized recursive types, that's a little sad. But I guess this is all going to be rolled into a unified version that mimics the head-spine form, so it won't be that apparent.

**Comment by effectfully** [comment]
> But I guess this is all going to be rolled into a unified version that mimics the head-spine form, so it won't be that apparent.

Oh, I should add the `nat` example, thanks. The dummy argument is completely invisible for the user indeed. But yeah, it's not a nice thing anyway.

**Comment by effectfully** [comment]
> I should add the `nat` example

Done.

**Comment by michaelpj** [comment]
I think the elaboration is correct, so let's update master.

**Pull request merged**